cli/tools/test/reporters/junit.rs

```diff
@@ -31,13 +36,21 @@ impl JunitTestReporter {
     match status {
       TestResult::Ok => quick_junit::TestCaseStatus::success(),
       TestResult::Ignored => quick_junit::TestCaseStatus::skipped(),
-      TestResult::Failed(failure) => quick_junit::TestCaseStatus::NonSuccess {
-        kind: quick_junit::NonSuccessKind::Failure,
-        message: Some(failure.overview()),
-        ty: None,
-        description: Some(failure.detail()),
-        reruns: vec![],
-      },
+      TestResult::Failed(failure) => {
+        let message = failure.overview();
+        let detail = failure.detail();
+        let message_stripped =
+          ANSI_ESCAPE_REMOVE.replace_all(message.as_str(), "");
+        let detail_stripped =
+          ANSI_ESCAPE_REMOVE.replace_all(detail.as_str(), "");
+        quick_junit::TestCaseStatus::NonSuccess {
+          kind: quick_junit::NonSuccessKind::Failure,
+          message: Some(message_stripped.into()),
+          ty: None,
+          description: Some(detail_stripped.into()),
+          reruns: vec![],
+        }
+      }
```

Comment:
I thought that it would be great to remove escape codes only when printing to non-tty, but considering it further, now I feel this isn't really necessary given that JUnit XML output is not read by human but actually ingested by some other programs in 99.9% cases.